### ISSUE TITLE
Detect asymmetric modeling

### ISSUE THREAD

**Issue by nikku** [bpmn-io/bpmnlint#95]
#### The rule should detect the following modeling patterns

Asymetric use of parallel / exclusive gateways, causing deadlocks or unintended token splits.

![image](https://user-images.githubusercontent.com/58601/212265723-a68b0819-2e1a-43f2-8374-b1a9295c8206.png)

![image](https://user-images.githubusercontent.com/58601/212265816-971513a5-31e0-42e6-bdf5-4b0893615c5d.png)

#### How does the rule improve the BPMN diagram?

Modeling symmetric is a modeling good practice. Not doing it leads to bad BPMN diagrams (harder to understand) and causes errors (dead locks, token splits).

#### Rule Details

* Name: `symmetric-gateways` 
* Default notification level: `error`

#### What alternatives did you consider? 

None.

**Comment by david-d-le** [comment]
Hi, I would like to solve this issue. However @nikku I would need some additional information.

What exactly asymmetric/symetric model means? Is it only that splitting/joining gateway must be of the same type? Must every gateway have it's pair? Are there other conditions that must meet?

**Comment by nikku** [comment]
@david-d-le The recommendation indeed is that every forking gateway has its joining pair. However in practice this is not feasible, so we want to carefully approach this, i.e. see if we can detect + compute good vs. bad patterns at all. 

I'm not sure due to the inherent complexity of this topic that it is a good community pick.

**Comment by david-d-le** [comment]
> @david-d-le The recommendation indeed is that every forking gateway has its joining pair. However in practice this is not feasible, so we want to carefully approach this, i.e. see if we can detect + compute good vs. bad patterns at all.
> 
> I'm not sure due to the inherent complexity of this topic that it is a good community pick.

Yes you are right. If every forking gateway has its joining pair it would be easy (I was using partitioning approach in my thesis which worked for finding the pairs).
But yes in practice where not every gateway has its pair, It is probably impossible to check so I will probably need to leave this to you.
